File: crates/ga-parser/tests/fixtures/kotlin-tiny/android-mini/UserService.kt
package com.example

import com.example.User
import com.example.UserRepository

class UserService {
    @Inject
    lateinit var userRepository: UserRepository

    fun getUser(id: Int): User {
        val cached = userRepository.findById(id)
        if (cached != null) return cached
        val u = User("user-$id")
        userRepository.save(u)
        return u
    }
}

fun String.isUserId(): Boolean = startsWith("user-")

annotation class Inject

File: crates/ga-parser/tests/fixtures/csharp-tiny/aspnet-mini/UserService.cs
using System;
using App.Models;
using App.Data;

namespace App.Services;

// Marker attribute used as Lang-C7 [Inject] sample (S-003c REFERENCES tests).
[AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
public sealed class InjectAttribute : Attribute { }

public class UserService
{
    [Inject]
    private UserRepository _userRepository;

    public User GetUser(int id)
    {
        var cached = _userRepository.FindById(id);
        if (cached != null) return cached;
        var u = new User($"user-{id}");
        _userRepository.Save(u);
        return u;
    }
}

// C# 9+ partial class — exercises AS-011 (multi-file partial merging at indexer level).
public partial class PartialDemo
{
    public void MethodA() { Console.WriteLine("a"); }
}

public partial class PartialDemo
{
    public void MethodB() { Console.WriteLine("b"); }
}
